lib/ts/recipe/session/sessionClass.ts

```diff
@@ -238,21 +246,31 @@ export default class Session implements SessionContainerInterface {
         return this.mergeIntoAccessTokenPayload(update, userContext);
     }
 
-    attachToRequestResponse(info: ReqResInfo) {
+    async attachToRequestResponse(info: ReqResInfo) {
         this.reqResInfo = info;
 
         if (this.accessTokenUpdated) {
-            const { res, transferMethod } = info;
+            const { req, res, transferMethod } = info;
 
-            setAccessTokenInResponse(res, this.accessToken, this.frontToken, this.helpers.config, transferMethod);
+            await setAccessTokenInResponse(
+                req,
+                res,
+                this.accessToken,
+                this.frontToken,
+                this.helpers.config,
+                transferMethod,
+                {}
```

Comment:
Maybe everywhere we default to using an empty object for user context we should consider adding a boolean flag indicating that the context is intentionally empty? Users may expect that the user context is never empty and this flag might help them handle it better. Might be worth discussing @rishabhpoddar